### ISSUE TITLE
Documentation missing for setting init options for openssh object construction

### ISSUE THREAD

**Issue by sdondley** [RexOps/Rex#1311]
The `Rex::Config->set_openssh_opt(%hash)` option lacks documentation for setting init options for an ssh object.

See https://github.com/RexOps/Rex/issues/1309 for reference.

**Comment by ferki** [comment]
Right, this is exactly what I just [commented](https://github.com/RexOps/Rex/issues/1309#issuecomment-626679960) over on #1309.

I hate to duplicate the same thing, but I think the best option is to copy-paste relevant parts of my comment here, and close the other issue as duplicate in favor of this one.

>The related user story format would probably look something like:
>
>**As:** a rex user,
**I would like to:** know how to set constructor options for the underlying Net::OpenSSH object,
**so:** I could fine-tune connection details according to my use case.
>
>This topic possibly could go directly into `Rex::Interface::Connection::OpenSSH`[...]
>
>For the list of available options, probably the best we could do is to point readers to the corresponding external documentation: Net::OpenSSH->new().
